<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Andrew S. Durden" />
  <meta name="author" content="H. Bernhard Schlegel" />
  <title>TDCI-CAP Manual</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <script src="/usr/share/javascript/mathjax/tex-mml-chtml.js" type="text/javascript"></script>
  <style>
  body {
    max-width: 1600px;
    width: 80%;
    margin: 0 auto;
  }
  table {
    border-collapse: collapse;
  }
  th, td {
    border: 1px solid #CCC;
    padding: 0.5em;
  }

  pre code {
    display: block;
    padding: 1em;
    margin: 0.5em 0;
    background-color: #f7f7f7;
    border: 1px solid #ddd;
    border-radius: 4px;
    font-family: "DejaVu Sans Mono", "Source Code Pro", monospace;
    overflow-x: auto;   /* allow horizontal scrolling if lines are long */
    line-height: 1.4;
  }
  code {
    background-color: #f7f7f7;
    border: 1px solid #ddd;
    padding: 0.1em 0.3em;
    border-radius: 3px;
    font-family: "DejaVu Sans Mono", monospace;
  }


  </style>

  <!-- Optionally load MathJax (if you want to control the version) -->
  <script type="text/javascript" async
    src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>
</head>
<body>
<header id="title-block-header">
<h1 class="title">TDCI-CAP Manual</h1>
<p class="author">Andrew S. Durden</p>
<p class="author">H. Bernhard Schlegel</p>
<p class="date">February 2025</p>
</header>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#theoretical-background">Theoretical Background</a>
<ul>
<li><a href="#hamiltonian">Hamiltonian</a></li>
<li><a href="#absorbing-potential">Absorbing Potential</a></li>
<li><a href="#electric-field-of-laser-pulse">Electric Field of Laser Pulse</a></li>
<li><a href="#wavefunctions">Wavefunctions</a></li>
<li><a href="#propagation-of-the-wavefunction">Propagation of the Wavefunction</a></li>
<li><a href="#ionization-rate">Ionization Rate</a></li>
<li><a href="#basis-sets">Basis Sets</a></li>
</ul></li>
<li><a href="#tutorial">Tutorial</a>
<ul>
<li><a href="#compilation">Compilation</a></li>
<li><a href="#gaussian-input">Gaussian Input</a></li>
<li><a href="#tdci-input">TDCI input</a></li>
<li><a href="#output-files">Output Files</a></li>
</ul></li>
</ul>
</nav>
<h1 id="theoretical-background">Theoretical Background</h1>
<p>The time-dependent configuration interaction (TDCI) code can be used to simulate the electronic structure of a molecule interacting with the electric field of an intense, ultra-short laser pulse. The time-independent, ground and excited configurations of a molecule in the absence of a field are combined with time-dependent linear coefficients to model the molecular wavefunction in the electric field of the laser pulse (the nuclear positions kept fixed). The interaction of the molecule with the electric field of the laser pulse is treated in the semiclassical dipole approximation and the pulses can be linear or elliptically polarized. For strong field ionization, the molecule is surrounded by a complex absorbing potential (CAP) to remove the outgoing electron (see Figure 1).</p>
<figure>
<img src="img/Figure1.png" alt="" /><figcaption><strong>Figure 1.</strong> Hydrogen atom wavefunction (light blue), Coulomb potential (dark blue) and complex absorbing potential (dashed dark blue) without a field. The strong field from a laser pulse suppresses the Coulomb potential (red) and the wavefunction (orange) can go over the Coulomb barrier or tunnel through it and be absorbed by the complex potential (dashed red).</figcaption>
</figure>
<p>For single ionization, the wavefunction includes the Hartree-Fock ground state and all singly excited configurations (CIS).<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>,<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> Ionization of cations can be modeled by a CISD-IP wavefunction which is built from singly ionized and singly excited, singly ionized configurations of the Hartree-Fock ground state of the neutral molecule. For molecules with heavy elements, spin-orbit coupling (SOC) can be included with an effective one electron SOC operator.<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> The wavefunction is propagated with the exponential of the field-dependent Hamiltonian using a Trotter factorization. The ionization yield for TDCI-CAP simulation is calculated from the decrease in the norm squared of the wavefunction as it interacts with the absorbing potential. The rate is computed from the expectation value of the absorbing potential using the time-dependent wavefunction. The integrals needed to construct the Hamiltonian are computed in an electronic structure program such as Gaussian. For strong field ionization standard molecular basis sets are augmented with several sets of diffuse functions to support the wavefunction as the electron leaves the molecule and interacts with the absorbing potential. The followings sections provide a brief overview of the theory and computational approaches of the TDCI code. More details on applications can be found in some of our papers.<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a><a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a><a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a><a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a></p>
<h2 id="hamiltonian">Hamiltonian</h2>
<p>In the TD-CI approach, the electronic wavefunction is propagated with the time-dependent Schrödinger equation:</p>
<p><span class="math display">\[\begin{equation}
i\hbar \frac{\partial}{\partial t} \Psi(t) \;=\; \hat{H}(t)\,\Psi(t) \;=\; 
\Bigl[\hat{H}_{el} + \hat{V}^{SOC} \;-\; \hat{\vec{\mu}} \cdot \vec{E}(t) \;-\; i \,\hat{V}^{abs}\Bigr] \,\Psi(t)
\tag{1}
\end{equation}\]</span></p>
<p><span class="math inline">\(\hat{H}_{el}\)</span> is the field-free, time-independent non-relativistic electronic Hamiltonian and the interaction with the electric field of an intense laser pulse is treated in the semiclassical dipole approximation, where <span class="math inline">\(\hat{\vec{\mu}}\)</span> is the dipole operator and <span class="math inline">\(\vec{E}\)</span> is the electric field. The spin-orbit coupling term is approximated by an effective one electron spin-orbit coupling operator.</p>
<p><span class="math display">\[\begin{equation}
\hat{V}^{SOC} \;=\; -\,\frac{\alpha_0^2}{2}\,\sum_A Z_A^{eff}\,\frac{(\vec{r} - \vec{r}_A)\,\times\,\nabla}{|\vec{r} - \vec{r}_A|^3}
\tag{2}
\end{equation}\]</span></p>
<p>Suitable values for <span class="math inline">\(Z^{eff}\)</span> have been reported by Koseki, Gordon and co-workers and by Chiodo and Russo. <span class="math inline">\(Z^{eff}\)</span> can be adjusted to reproduce the experimentally observed spin-orbit splitting. <span class="math inline">\(\hat{V}^{abs}\)</span> is the absorbing potential.</p>
<h2 id="absorbing-potential">Absorbing Potential</h2>
<p>The absorbing potential, <span class="math inline">\(\hat{V}^{abs}\)</span>, surrounds the molecule starting at a distance of about 10 – 20 bohr and removes the outgoing part of the wavefunction, thereby simulating ionization.<br />
The absorbing potential for the molecule is constructed from spherical potentials centered on each atom.<br />
The potential on atom C is written in terms of the distance from the nucleus, <span class="math inline">\(r_c = |\vec{r} - \vec{C}|\)</span>.<br />
In the quadratic form, the atomic potential starts at <span class="math inline">\(R_A\)</span>, rises quadratically to <span class="math inline">\(\frac{R_A+R_B}{2}\)</span>, turns over quadratically to <span class="math inline">\(R_B\)</span>, and is constant, <span class="math inline">\(V_{max}\)</span>, beyond <span class="math inline">\(R_B\)</span>:</p>
<p><span class="math display">\[\begin{equation}
\hat{V}^{abs}_{C}(r_c) \;=\;
\begin{cases} 
0, &amp; r_c \le R_A,\\[6pt]
V_{max}, &amp; r_c \ge R_B,\\[6pt]
2V_{max}\Bigl(\frac{r_c - R_A}{R_B - R_A}\Bigr)^2, &amp; R_A \le r_c \le \tfrac{R_A + R_B}{2},\\[6pt]
V_{max} \;-\; 2V_{max}\Bigl(\frac{R_B - r_c}{R_B - R_A}\Bigr)^2, &amp; \tfrac{R_A + R_B}{2} \le r_c \le R_B.
\end{cases}
\tag{3}
\end{equation}\]</span></p>
<p>In the <span class="math inline">\(\sin^2\)</span> form, the atomic potential on an atom rises as <span class="math inline">\(\sin^2\)</span> between <span class="math inline">\(R_A\)</span> and <span class="math inline">\(R_B\)</span>,</p>
<p><span class="math display">\[\begin{equation}
\hat{V}^{abs}_{C}(r_c) \;=\; 
V_{max}\,\sin^2\Bigl(\tfrac{\pi}{2}\,\tfrac{r_c - R_A}{R_B - R_A}\Bigr), 
\quad R_A \,\le\, r_c \,\le\, R_B.
\tag{4}
\end{equation}\]</span></p>
<p>The absorbing potential for the molecule is equal to the minimum of the values of the atomic absorbing potentials. This has also been termed a Voronoi absorbing potential.</p>
<p><span class="math display">\[\begin{equation}
\hat{V}^{abs}(\vec{r}) \;=\; 
\min\Bigl(\hat{V}^{abs}_1(r_1),\;\hat{V}^{abs}_2(r_2),\;\dots,\;\hat{V}^{abs}_{\text{Natoms}}(r_{\text{Natoms}})\Bigr).
\tag{5}
\end{equation}\]</span></p>
<p>Typical values of the parameters for the absorbing potential are <span class="math inline">\(R_A = 3.5\)</span> times the van der Waals radius for each atom (so that the absorbing potential starts far enough beyond the Coulomb well), <span class="math inline">\(R_B = 10\)</span> times the van der Waals radius (so that the rise in the absorbing potential is gradual enough to minimize any reflection) and <span class="math inline">\(V_{\text{max}} = 10\)</span> hartree (so that the potential is strongly absorbing but the integrals remain finite).<br />
Figure 2 shows examples of atomic and molecular absorbing potentials.</p>
<figure>
<img src="img/Figure2.png" alt="" /><figcaption><strong>Figure 2.</strong> (a) Shape of an atomic absorbing potential (<span class="math inline">\(\sin^2\)</span> form in red, quadratic form in dashed blue), (b) 2D plot of <span class="math inline">\(R_A\)</span> for absorbing potential of carbon (black), oxygen (red) and hydrogen (grey) in <span class="math inline">\(CH_2O\)</span> (c) 3D plot of <span class="math inline">\(R_A\)</span> (gold) and <span class="math inline">\(\tfrac{R_A+R_B}{2}\)</span> (blue) for the molecular absorbing potential for <span class="math inline">\(CH_2O\)</span>.</figcaption>
</figure>
<h2 id="electric-field-of-laser-pulse">Electric Field of Laser Pulse</h2>
<p>The electric field for a linearly polarized <span class="math inline">\(n\)</span>-cycle pulse with a <span class="math inline">\(\sin^2\)</span> envelope is:</p>
<p><span class="math display">\[\begin{equation}
E(t) \;=\;
\begin{cases}
E_{max}\,\sin^2\!\Bigl(\tfrac{\omega t}{2n}\Bigr)\,\cos(\omega t), &amp; 0 \le \omega t \le 2\pi n,\\[6pt]
0, &amp; \omega t &gt; 2\pi n.
\end{cases}
\tag{6}
\end{equation}\]</span></p>
<p>For an <span class="math inline">\(n\)</span>-cycle circularly polarized 800 nm pulse in the <span class="math inline">\(xz\)</span>-plane with a <span class="math inline">\(\sin^2\)</span> envelope:</p>
<p><span class="math display">\[\begin{equation}
E_x(t) \;=\; E_{max}\,\sin^2\!\Bigl(\tfrac{\omega t}{2n}\Bigr)\,\Bigl[-\,\cos(\omega t)\,\cos(\gamma) \;-\; \sin(\omega t)\,\sin(\gamma)\Bigr],
\end{equation}\]</span></p>
<p><span class="math display">\[\begin{equation}
E_z(t) \;=\; E_{max}\,\sin^2\!\Bigl(\tfrac{\omega t}{2n}\Bigr)\,\Bigl[\cos(\omega t)\,\sin(\gamma) \;-\; \sin(\omega t)\,\cos(\gamma)\Bigr],
\end{equation}\]</span></p>
<p><span class="math display">\[\begin{equation}
E_x(t) \;=\; E_z(t) \;=\; 0,\quad \omega t \ge 2\pi n.
\tag{7}
\end{equation}\]</span></p>
<p><span class="math inline">\(E_{max}\)</span> is the maximum value for the electric field and <span class="math inline">\(\gamma\)</span> is the Euler angle that determines the direction of the major axis of an elliptically polarized pulse.<br />
To obtain directional information for ionization, a static field can be used instead of an oscillating field.<br />
To avoid non-adiabatic excitations, the electric field slowly ramps up and turns over to a constant value:</p>
<p><span class="math display">\[\begin{equation}
E(t) \;=\;
\begin{cases}
E_{max}\Bigl(1 - \Bigl(1 - \tfrac{t}{t_{ramp}}\Bigr)^4\Bigr), &amp; 0 \le t \le t_{ramp},\\[6pt]
E_{max}, &amp; t &gt; t_{ramp}.
\end{cases}
\tag{8}
\end{equation}\]</span></p>
<p>The shapes of a 7-cycle linear pulse and a static pulse are shown in Figure 3.<br />
Several other pulse shapes available in the TDCI code are described in the Appendix.</p>
<figure>
<img src="img/Figure3.png" alt="" /><figcaption><strong>Figure 3.</strong> (a) A 7-cycle linearly polarized 800 nm pulse with a <span class="math inline">\(\sin^2\)</span> envelope (Eq. 6), (b) 2-cycle circularly polarized 800 nm pulse with a <span class="math inline">\(\sin^2\)</span> envelope showing <span class="math inline">\(x\)</span> and <span class="math inline">\(z\)</span> components in green and red, respectively (Eq. 7), and (c) a “static” pulse used to probe the angular dependence of strong field ionization (Eq. 8).</figcaption>
</figure>
<h2 id="wavefunctions">Wavefunctions</h2>
<p>In the TD-CI approach, the wavefunction is written as a linear combination of time-dependent coefficients with time-independent states of the field-free Hamiltonian, <span class="math inline">\(\Psi_s\)</span>:</p>
<p><span class="math display">\[\begin{equation}
\Psi(t) \;=\; \sum_{s}\,C_{s}(t)\,\Psi_s
\tag{9}
\end{equation}\]</span></p>
<p>For single ionization of a neutral molecule to cations, a wavefunction consisting of the ground state and all singly excited configurations (CIS) is suitable,</p>
<p><span class="math display">\[\begin{equation}
\Psi(t) \;=\; c_0(t)\,\psi_0 \;+\; \sum_{i,a}\,c_i^a(t)\,\psi_i^a
\tag{10}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(\psi_i^a\)</span> are singly excited determinants and <span class="math inline">\(i, j\)</span> etc. and <span class="math inline">\(a, b\)</span> etc. are indices for the occupied and virtual molecular spin-orbitals, respectively. For ionization of a cation to a dication, a spin-unrestricted wavefunction could be used, but this treats the <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> spin-orbitals differently. This problem can be overcome by using a CISD-IP wavefunction. The wavefunction is constructed using the molecular orbitals of the closed shell system and includes singly ionized determinants, <span class="math inline">\(\psi_x\)</span>, and singly excited, singly ionized determinants, <span class="math inline">\(\psi_{xi}^a\)</span>:</p>
<p><span class="math display">\[\begin{equation}
\Psi(t) \;=\; \sum_{x}\,c_x(t)\,\psi_x \;+\; \sum_{x,i,a}\,c_{xi}^a\,\psi_{xi}^a
\tag{11}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(x\)</span> indicates the ionized molecular spin-orbitals. The ionizations generate the ground and excited states of the cation, and the single excitations serve to improve the energies of these cation states. As a result, the energies of the cation states will depend slightly on the number of virtual orbitals used to construct the singly excited, singly ionized determinants.</p>
<h2 id="propagation-of-the-wavefunction">Propagation of the Wavefunction</h2>
<p>For typical strong field ionization studies, numerous simulations of 20 – 50 femtoseconds are needed (e.g., different pulse shapes, intensities and directions, different delay time in pump-probe simulations, etc.). Because of the strong fields and the wide range of the energies of the excited configurations, small time steps are required for the propagation. In the TDCI code the wavefunction is propagated with the exponential of the Hamiltonian, since this unitary transformation is very stable and allows for larger time steps. Typically, a time step of 0.05 au = 1.2 attoseconds can be used. Reducing the time step by a factor of 2 changes the ionization rate by less than 0.02% for typical simulations. The exponential of the Hamiltonian can be obtained via a Trotter factorization with components depending on the field-free Hamiltonian, the absorbing potential, and the dipole moment matrix.</p>
<p>For a linearly polarized pulse, the propagation for a time step <span class="math inline">\(\Delta t\)</span> is</p>
<p><span class="math display">\[\begin{equation}
\Psi(t + \Delta t) \;=\; \exp\bigl(-\,i\,\hat{H}\,\Delta t\bigr)\,\Psi(t)
\tag{12}
\end{equation}\]</span></p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
C(t + \Delta t) \;=\; 
&amp;\exp\Bigl(-\,i\,\hat{H}_{el}\,\tfrac{\Delta t}{2}\Bigr)\,
\exp\Bigl(-\,V^{abs}\,\tfrac{\Delta t}{2}\Bigr)\, \\
&amp;\exp\Bigl(i\,E\bigl(t + \tfrac{\Delta t}{2}\bigr)\,D\,\Delta t\Bigr)\, \\
&amp;\exp\Bigl(-\,V^{abs}\,\tfrac{\Delta t}{2}\Bigr)\,
\exp\Bigl(-\,i\,\hat{H}_{el}\,\tfrac{\Delta t}{2}\Bigr)\,C(t)
\end{aligned}
\tag{13}
\end{equation}\]</span></p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
C(t + \Delta t) \;=\; 
&amp;\exp\Bigl(-\,i\,\hat{H}_{el}\,\tfrac{\Delta t}{2}\Bigr)\,U^{T}\, \\
&amp;\exp\Bigl(i\,E\bigl(t + \tfrac{\Delta t}{2}\bigr)\,d\,\Delta t\Bigr)\, \\
&amp;U\,\exp\Bigl(-\,i\,\hat{H}_{el}\,\tfrac{\Delta t}{2}\Bigr)\,C(t)
\end{aligned}
\tag{14}
\end{equation}\]</span></p>
<p>This requires some initial diagonalizations but avoids the exponentiation of a full matrix at every time step. The field-free Hamiltonian is time-independent and can be diagonalized once at the beginning of the simulation. By working in the eigenbasis of the field-free Hamiltonian, <span class="math inline">\(\exp(-i\,\hat{H}_{el}\,\Delta t / 2)\)</span> is a diagonal matrix and is easy to calculate. Because the absorbing potential is time-independent, <span class="math inline">\(\exp(-V^{abs}\,\Delta t / 2)\)</span> needs to be calculated only once. The calculation of <span class="math inline">\(\exp(i\,\vec{E}(t + \Delta t/2)\,\mathbf{D}\,\Delta t)\)</span> would require an exponentiation of a full matrix at each time step. However, by diagonalizing <span class="math inline">\(\mathbf{d} = \mathbf{W}^T\,\mathbf{D}\,\mathbf{W}\)</span> once at the beginning of the simulation and working in the eigenbasis of <span class="math inline">\(\mathbf{D}\)</span>, the contribution reduces to an exponential of a <strong>time-dependent diagonal matrix</strong>, <span class="math inline">\(\exp(i\,\vec{E}(t + \Delta t/2)\,\mathbf{d}\,\Delta t)\)</span>. The product <span class="math inline">\(\mathbf{U} = \exp(-V^{abs}\,\Delta t/2)\,\mathbf{W}^T\)</span> is formed once at the beginning of the propagation. Thus, all of the <span class="math inline">\(N^3\)</span> steps need to be done only once at the beginning and can be reused for many subsequent simulations. A propagation step for a linearly polarized pulse with fixed nuclear positions scales as <span class="math inline">\(N^2\)</span> and involves two full matrix-vector multiplies (<span class="math inline">\(\mathbf{U}\)</span> and <span class="math inline">\(\mathbf{U}^T\)</span>) and three diagonal matrix-vector multiplies (<span class="math inline">\(\exp(-i\,\hat{H}_{el}\,\Delta t/2)\)</span> and <span class="math inline">\(\exp(i\,\vec{E}(t + \Delta t/2)\,\mathbf{d}\,\Delta t)\)</span>).</p>
<p>The corresponding Trotter factorization for a circularly polarized pulse involves two oscillating fields:</p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
C(t + \Delta t) = &amp; \exp(-i \hat{H}_{el} \Delta t / 2) \exp(-V^{abs} \Delta t / 2) \\
&amp; \mathbf{W}_2^T \exp\left(i E_z(t + \Delta t / 2) d_2 \Delta t / 2\right) \mathbf{W}_2 \\
&amp; \mathbf{W}_1^T \exp\left(i E_x(t + \Delta t / 2) d_1 \Delta t\right) \mathbf{W}_1 \\
&amp; \mathbf{W}_2^T \exp\left(i E_z(t + \Delta t / 2) d_2 \Delta t / 2\right) \mathbf{W}_2 \\
&amp; \exp(-V^{abs} \Delta t / 2) \exp(-i \hat{H}_{el} \Delta t / 2) C(t).
\end{aligned}
\tag{15}
\end{equation}\]</span></p>
<p>Here, <span class="math inline">\(\mathbf{W}_1\,\mathbf{D}_1\,\mathbf{W}_1^T = d_1\)</span> and <span class="math inline">\(\mathbf{W}_2\,\mathbf{D}_2\,\mathbf{W}_2^T = d_2\)</span> are the eigenvalues and eigenvectors of the transition dipole matrices <span class="math inline">\(\mathbf{D}_1\)</span> and <span class="math inline">\(\mathbf{D}_2\)</span> in the two orthogonal field directions. The product <span class="math inline">\(\mathbf{U}&#39; = \mathbf{W}_1\,\mathbf{W}_2\)</span> is formed once at the beginning of the propagation. A propagation step for a circularly polarized pulse with fixed nuclei involves four full matrix-vector multiplies (<span class="math inline">\(\mathbf{U}\)</span>, <span class="math inline">\(\mathbf{U}^T\)</span>, <span class="math inline">\(\mathbf{U}&#39;\)</span>, <span class="math inline">\(\mathbf{U}^{\prime T}\)</span>) and five diagonal matrix-vector multiplies.</p>
<h2 id="ionization-rate">Ionization Rate</h2>
<p>The ionization rate is taken as the rate of decrease in the norm squared of the wavefunction as the wavefunction interacts with the absorbing potential (Figure 4).</p>
<figure>
<img src="img/Figure4.png" alt="" /><figcaption><strong>Figure 4.</strong> Strong field ionization of hydrogen atom (a) electric field of a “static” pulse with three different intensities (b) dipole moment demonstrating hydrogen responds adiabatically to the field (c) decrease in the norm of the wavefunction as hydrogen is ionized by the electric field.</figcaption>
</figure>
<p>Using the time-dependent Schrödinger equation,</p>
<p><span class="math display">\[\begin{equation}
i\,\frac{\partial\,\Psi(t)}{\partial t} \;=\; \Bigl[\hat{H}(t) \;-\; i\,\hat{V}^{\text{abs}}\Bigr]\,\Psi(t),
\tag{16}
\end{equation}\]</span></p>
<p>the rate can be related to the expectation value of the absorbing potential:</p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
\text{rate}(t)
&amp;=\; -\,\frac{1}{ \langle \Psi(t)\,|\,\Psi(t)\rangle} \; \frac{d\,\langle \Psi(t)\,|\,\Psi(t)\rangle}{dt} \\[6pt]
&amp;=\; -\,\frac{\Bigl\langle \Psi(t)\,\Big|\, \frac{d\,\Psi(t)}{dt} \Bigr\rangle}{ \langle \Psi(t)\,|\,\Psi(t)\rangle} \;+\;\text{(c.c.)} \\[6pt]
&amp;=\; -\,\frac{\Bigl\langle \Psi(t)\,\Big|\,-\,i\,\bigl[\hat{H}(t) \;-\; i\,\hat{V}^{\text{abs}}\bigr]\,\Psi(t)\Bigr\rangle}{ \langle \Psi(t)\,|\,\Psi(t)\rangle} \;+\;\text{(c.c.)} \\[6pt]
&amp;=\; 2\,\frac{\Bigl\langle \Psi(t)\,\Big|\,\hat{V}^{\text{abs}}\,\Big|\,\Psi(t)\Bigr\rangle}{ \langle \Psi(t)\,|\,\Psi(t)\rangle} \\[4pt]
&amp;=\; 2\,\sum_{r,s}\,C_r^*(t)\,C_s(t)\,\langle \,\Psi_r\,|\,\hat{V}^{\text{abs}}\,|\,\Psi_s\,\rangle,\quad \sum_{r,s}\,C_r^*(t)\,C_s(t)=1 .
\end{aligned}
\tag{17}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(r\)</span> and <span class="math inline">\(s\)</span> are indices for configurations. For simplicity, the CIS wavefunction is used in the following equations, but these can be easily extended to CISD-IP. In terms of determinants, the ionization rate is</p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
\text{rate}(t)
=\;2\Bigl[\; c_0^*(t)\,c_0(t)\,\langle \Psi_0 | \hat{V}^{\text{abs}} | \Psi_0 \rangle
\;+\;\sum_{i,a}\,c_i^{a*}(t)\,c_0(t)\,\langle \Psi_i^a | \hat{V}^{\text{abs}} | \Psi_0 \rangle \\[6pt]
\quad\quad+\;\sum_{j,b}\,c_0^*(t)\,c_j^b(t)\,\langle \Psi_0 | \hat{V}^{\text{abs}} | \Psi_j^b \rangle
\;+\;\sum_{i,j,a,b}\,c_i^{a*}(t)\,c_j^b(t)\,\langle \Psi_i^a | \hat{V}^{\text{abs}} | \Psi_j^b \rangle
\Bigr].
\end{aligned}
\tag{18}
\end{equation}\]</span></p>
<p>The rate can also be written in terms of the density matrix and the absorbing potential in the basis of the molecular orbitals, <span class="math inline">\(\phi_p\)</span>:</p>
<p><span class="math display">\[\begin{equation}
\text{rate}(t)
\;=\; 2\,\sum_{p,q}^{\text{all}}\;\rho_{pq}(t)\;\langle \phi_p\,|\,\hat{V}^{\text{abs}}\,|\,\phi_q \rangle,
\tag{19}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(\rho_{pq}(t)\)</span> is the time-dependent one-particle density matrix derived from the CI wavefunction, and indices <span class="math inline">\(p\)</span> and <span class="math inline">\(q\)</span> run over all occupied and virtual orbitals.<br />
This form is also applicable to rt-TD-DFT.</p>
<h2 id="basis-sets">Basis Sets</h2>
<p>For simulations of strong field ionization with TDCI, a standard molecular basis set must be augmented by several sets of diffuse functions to support the wavefunction as it is distorted by the strong field and interacts with the absorbing potential. Various sets of diffuse functions have been developed and evaluated for their ability to treat strong field ionization. The added diffuse functions should represent the unbound electron density optimally as it interacts with the laser field and the absorbing potential but should be limited in number so that the simulations can be carried out efficiently. These sets include diffuse <em>s</em>, <em>p</em>, <em>d</em>, and <em>f</em> gaussian functions with selected even-tempered exponents of the form 0.0001×<span class="math inline">\(2^n\)</span> placed on each atom. When combined with the aug-cc-pVTZ molecular basis set and an absorbing potential starting at 3.5 times the van der Waals radius for each atom, the least diffuse <em>s</em>, <em>p</em> and <em>f</em> functions should have exponents of 0.0256 and 0.0512 for <em>d</em> functions. The most diffuse <em>s</em>, <em>p</em>, <em>d</em>, and <em>f</em> functions should have exponents of 0.0032, 0.0032, 0.0064 and 0.0064, respectively (or possibly smaller). Because diffuse functions on adjacent centers overlap strongly, the exponents should not be too small since this leads to severe linear dependencies and SCF convergence problems.</p>
<figure>
<img src="img/Figure5.png" alt="" /><figcaption><strong>Figure 5.</strong> Example of diffuse s, p, d and f gaussian basis functions added to support the wavefunction in the region between the Coulomb well and the absorbing potential.</figcaption>
</figure>
<h1 id="tutorial">Tutorial</h1>
<h2 id="compilation">Compilation</h2>
<p>TDCI-CAP can be compiled with <strong>nvfortran</strong> on Unix systems supported by NVIDIA’s HPC SDK. We have tested only CentOS, Ubuntu, and Archlinux x64 builds.</p>
<p>The NVIDIA HPC SDK can be downloaded at<br />
<a href="https://developer.nvidia.com/hpc-sdk-downloads" class="uri">https://developer.nvidia.com/hpc-sdk-downloads</a><br />
NVIDIA’s install instructions:<br />
<a href="https://docs.nvidia.com/hpc-sdk/hpc-sdk-install-guide/index.html" class="uri">https://docs.nvidia.com/hpc-sdk/hpc-sdk-install-guide/index.html</a></p>
<p>Once the environment is prepared, you can clone the repository and compile the code:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="fu">git</span> clone https://github.com/hbschlegel/TDCI-CAP</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="bu">cd</span> TDCI-CAP</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="fu">make</span></span></code></pre></div>
<p>The first time TDCI-CAP is built, it may take several minutes to compile its dependencies hdf5 and makedepf90. After compilation completes, you will find the <code>tdci</code> executable in the <code>bin/</code> subdirectory.</p>
<h2 id="gaussian-input">Gaussian Input</h2>
<p>Before running TDCI, we must run a preliminary Gaussian Configuration Interaction Singles (CIS) calculation to obtain the necessary integrals. The one-electron integrals are written in the atomic orbital basis and include the dipole moment, spin-orbit, and absorbing potential integrals (<span class="math inline">\(Z_{\text{eff}}\)</span> needs to be specified for each atom for spin-orbit calculations, see Nuclei Props in in the <a href="https://gaussian.com/molspec/">molecule specification</a> section). The transformed two-electron integrals are written in the molecular orbital basis in the Gaussian “bucket” format. Since most strong field simulations use only a limited number of occupied orbitals and eliminate the highest energy virtual orbitals, the Gaussian input needs to specify the number of occupied and virtual orbitals for the integral transformation. These files can be very large but can be used for multiple simulations as long as the molecular geometry and basis set are the same. We use Gaussian development version gdvj28p.<br />
Below is an example Gaussian input file. Several inputs options are necessary and are described in more detail below:</p>
<pre><code>%chk=h2o.chk
%mem=10GB
%nproc=10
#P ucis(mo,nstates=5)/aug-cc-pVDZ extrabasis gfprint Pop=Full int(acc2e=12) 
IOp(3/194=10003501,3/195=10,6/18=1,8/10=91,8/37=2,8/38=-10,9/127=2)
OUTPUT=(faf,i4lab,Files=(619,751,870,1,2,3,4,5,6,7,8,9,10,11,12,13,14))

h2o

0 1
O        0.000000    0.000000    0.110812
H        0.000000   -0.783976   -0.443248
H        0.000000    0.783976   -0.443248

O H 0
 s 1 1.00
 .0128000000D-00 1.0000000000D+00
 s 1 1.00
 .0064000000D-00 1.0000000000D+00
 p 1 1.00
 .0128000000D-00 1.0000000000D+00
****

MatrixElements.faf

</code></pre>
<p><code>IOp(3/194)</code> sets the absorbing potential parameters proportional to each atom’s van der Waals radius in the following format: <code>bbbaaa01</code> Where the inner boundary (<span class="math inline">\(R_{\text{A}}\)</span>) is set at aaa/10 times the atom’s van der Waals radius, and the outer boundary (<span class="math inline">\(R_{\text{B}}\)</span>) is set at bbb/10 times the atom’s van der Waals radius. For example, the <code>3/194=10003501</code> in the example sets <span class="math inline">\(R_{\text{A}}\)</span> to 3.5 radii, and <span class="math inline">\(R_{\text{B}}\)</span> to 10 radii.</p>
<p><code>IOp(3/195)</code> sets the maximum value of the absorbing potential (<span class="math inline">\(V_{\text{max}}\)</span>) in atomic units. 10 is suitable.</p>
<p><code>IOp(6/18)=1</code> and <code>IOp(9/127)=2</code> are necessary for calculating some of the matrix elements that are written to MatrixElements.faf</p>
<p><code>IOp(8/10)=91</code> enables <code>IOp(8/37)</code> and <code>IOp(8/38)</code>, which specify the active space.</p>
<p><code>IOp(8/37)=N</code> selects orbital N as the first occupied orbital to be included in the active space. Excitations out of orbitals below will not be considered. Note tIn Gaussian the orbitals are indexed staring at 1, not 0.</p>
<p><code>IOp(8/38)=N</code> If N is positive, selects orbital N as the last orbital to be included in the active space. If N is negative, the N highest energy virtual orbitals will be removed from the active space. In practice, we select this to ignore orbitals with energies above about 3.0 Hartree.</p>
<p>Don’t change the line that starts with <code>OUTPUT=</code>; it specifies the indices of the various files to be included in the MatrixElements.faf file.</p>
<p>When selecting an active space, one must be careful to ensure degenerate orbitals remain in the same active or non-active set. If one degenerate orbital is left out of the active space while others are included, it will cause unphysical results.</p>
<p>If you’re having convergence difficulties, try combinations of these parameters:</p>
<blockquote>
<p><code>SCF(QC,NoVarAcc,NoIncFock,MaxCYC=64,VShift=500)</code></p>
</blockquote>
<p>For more information on the IOps, consult the <a href="https://gaussian.com/iops/">Gaussian overlay documentation</a>. The SCF parameters are documented <a href="https://gaussian.com/scf/">here</a>.</p>
<p>NOTE: If you are using shared computing like WSU’s Grid, please remember to follow their rules and policies. This means you should run Gaussian and TDCI from an sbatch script that is scheduled through SLURM. See your computing resource’s documentation for more details.</p>
<p>Once your Gaussian calculation successfully finishes, there should be a MatrixElements.faf in the Gaussian job directory.</p>
<h2 id="tdci-input">TDCI input</h2>
<p>TDCI expects both <code>MatrixElements.faf</code> and <code>input</code> to be in the current working directory. Sample TDCI input files can be found in the test/tests folder, or in the section below. To perform a TDCI simulation, simply execute the tdci binary (or sbatch script) from inside the job directory. We recommend separating the Gaussian and TDCI job directories, and creating a <a href="https://en.wikipedia.org/wiki/Symbolic_link">symbolic link (symlink)</a> to the MatrixElements.faf in your TDCI job directory. Here’s how to set up a TDCI job directory in the recommended way. We assume that the Gaussian calculation is performed inside a directory called <code>gaussian_h2o</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="fu">mkdir</span> tdci_h2o                                # Create TDCI job directory</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="bu">cd</span> tdci_h2o                                   # Enter TDCI job directory</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a><span class="fu">ln</span> -s ../gaussian_h2o/MatrixElements.faf .    # Create symlink.</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a><span class="fu">ls</span> -alt                                       # Print out a directory listing</span></code></pre></div>
<p>Once you finish setting up your input file, your directory should look something like this:</p>
<pre><code>drwxr-xr-x 18 user user     4096 Sep 23 16:47 ..
drwxr-xr-x 18 user user     4096 Sep 23 16:47 .
-rw-r--r--  1 user user     2052 Sep 22 18:42 input
lrwxrwxrwx  1 user user       39 Sep 22 18:42 MatrixElements.faf -&gt; ../gaussian_h2o/MatrixElements.faf</code></pre>
<p>Then you can run TDCI simulation. If you are running on your own hardware, you can directly invoke the executable:</p>
<pre><code>~/TDCI-CAP/bin/tdci</code></pre>
<p>If you are using shared computing, make sure to wrap this in an sbatch script.</p>
<!-- Commenting out this section so we don't have to explain write_matrices = .true.

 TDCI itself exports some data tables in the job directory, but TDCI is also packaged with a python analysis script that generates many tables and plots.
The tools/rate\_analyzer.py script must be executed from the job directory, and passed the path of the gaussian output log as an argument.
```
~/tdci_h2o$ python3 ~/TDCI-CAP/tools/rate_analyzer.py ../gaussian_h2o/h2o.log
```
-->
<p>Next, we provide a sample tdci <code>input</code> file. The <code>input</code> file is separated into “namelist” sections that start with <code>&amp;</code> and end with <code>/</code>.</p>
<p>A table with descriptions of each parameter is provided below.</p>
<p>In this example, we set the initial wavefunction as a superposition between the ground and second excited states with the <code>init_states</code> and <code>init_coeffs</code> parameters.</p>
<p>The <code>FIELD</code> section specifies that a “static” field (slowly ramps up to a static value) will be applied to the system.</p>
<p>The <code>FIELD_strengths</code> section specifies that we will only propagate one field strength of 0.0500 atomic units.</p>
<p>The <code>FIELD_directions</code> section specifies that we will apply this field in two different directions, the <span class="math inline">\((\theta,\phi)=(0,0)\)</span> direction, and the <span class="math inline">\((\theta,\phi)=(30,0)\)</span> direction.</p>
<p>The parameters in the <code>SYSTEM</code> section control the propagation scheme, including the duration and number of timesteps.</p>
<p>The <code>hdf5</code> section requests and h5 file be written, and that density matrices be included.</p>
<pre><code> &amp;DYNAMICS
 init_states(0) = 2
 init_states(1) = 1
 init_states(2) = 3
 init_coeffs(1) = (1.00000,0.00000)
 init_coeffs(2) = (1.00000,0.00000)
 restart        = .false.
 Qsave          = .false.
 /
 &amp;FIELD_units
 omega_units = &#39;au&#39;
 /
 &amp;FIELD
 dirform  = &#39;polar&#39;
 envelope = &#39;stat&#39;
 /
&amp;FIELD_strengths 
 nemax =       1
 read_emax(1) = 0.0500
 /
&amp;FIELD_directions
 ndir = 2 
 read_theta( 1) = 0.d0   ;  read_phi( 1) = 0.d0 
 read_theta( 2) = 30.d0  ;  read_phi( 2) = 0.d0 
/
 &amp;SYSTEM_units
 dt_units     = &#39;au&#39; 
 eigmax_units = &#39;au&#39; 
 /
 &amp;SYSTEM
 dt          =  0.050
 eigmax      = 10.000
 ionization  = -1.000
 jobtype     =  &#39;cis&#39;
 nstep       =   1000
 outstep     =    100
 nactive     =      0
 nvirtual    =      0
 socfac      = 1.00000000
 socfacZ     = 1.00100000
 IP_alpha_beta = .false.
 QeigenDC = .true.
 /
 &amp;InOutFILES
 Qread_tdcidata  = .false.
 tdcidatfile     = &#39;MatrixElements.faf&#39;
 outputfile      = &#39;OUTPUT&#39;
 restartbinfile  = &#39;OUTPUT_RESTART.bin&#39;
 Qmo_dens        = .false.
 /
 &amp;hdf5
 h5inc_enable = .true.
 h5inc_density = .true.
 h5inc_psi = .false.
 h5inc_psi_det0 = .false.
 /
</code></pre>
<p>Table of input parameters:</p>
<table>
<colgroup>
<col style="width: 5%" />
<col style="width: 5%" />
<col style="width: 31%" />
<col style="width: 12%" />
<col style="width: 44%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Namelist</strong></th>
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
<th><strong>Allowed values</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DYNAMICS</td>
<td>init_states(i)</td>
<td>init_state(0) is the number of initial states. Init_state(i) for i&gt;0 is the state index of state i, where 1 is the ground state.</td>
<td>N/A</td>
<td></td>
</tr>
<tr class="even">
<td>DYNAMICS</td>
<td>init_coeffs(i)</td>
<td>Initial coefficient of state i</td>
<td>Complex (e.g., (1.00, 0.00))</td>
<td>Example: Init_coeffs(1) = (1.0, 0.0)</td>
</tr>
<tr class="odd">
<td>DYNAMICS</td>
<td>restart</td>
<td>Reads RESTART.bin to restart the propagation</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="even">
<td>DYNAMICS</td>
<td>Qsave</td>
<td>Writes RESTART.bin</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="odd">
<td>FIELD_units</td>
<td>omega_units</td>
<td>Units for the omega parameter</td>
<td>‘au’</td>
<td>‘au’ angular frequency in atomic units \ ‘nm’ wavelength in nanometers</td>
</tr>
<tr class="even">
<td>FIELD</td>
<td>dirform</td>
<td>Coordinate system used for FIELD_directions namelist.</td>
<td>‘polar’, ‘cart’</td>
<td>‘polar’: Polar coordinates \ ‘cart’: Cartesian coordinates</td>
</tr>
<tr class="odd">
<td>FIELD</td>
<td>ellipt</td>
<td>Ellipticity for circularly polarized fields</td>
<td>Float from 0.0–1.0</td>
<td>A value of 1.0 corresponds to a circular pulse.</td>
</tr>
<tr class="even">
<td>FIELD</td>
<td>envelope</td>
<td>Envelope that outlines the pulse</td>
<td>‘none’, ‘cos2’, ‘trap’, ‘gaus’, ‘stat’, ‘band’, ‘ramp’, ‘sin2’, ‘cirl’, ‘cirr’</td>
<td>For a full description of the options, see the relevant section in documentation.</td>
</tr>
<tr class="odd">
<td>FIELD</td>
<td>ncyc</td>
<td>Number of optical cycles that are enclosed by the envelope</td>
<td>Positive integer</td>
<td>Does not apply to fields ‘none’, ‘static’, and ‘ramp’.</td>
</tr>
<tr class="even">
<td>FIELD</td>
<td>omega</td>
<td>Frequency of the pulse</td>
<td>Positive float</td>
<td>Units are set by the omega_units parameter.</td>
</tr>
<tr class="odd">
<td>FIELD</td>
<td>phase</td>
<td>Carrier envelope phase (CEP) in degrees</td>
<td>float</td>
<td></td>
</tr>
<tr class="even">
<td>FIELD</td>
<td>euler</td>
<td>Sets the Euler angle (gamma) for elliptical pulses.</td>
<td>float</td>
<td></td>
</tr>
<tr class="odd">
<td>FIELD_strengths</td>
<td>nemax</td>
<td>Number of propagations to perform with different field strengths</td>
<td>Positive integer</td>
<td>Read_emax(i) must be set for i=1,…,nemax</td>
</tr>
<tr class="even">
<td>FIELD_strengths</td>
<td>read_emax(i)</td>
<td>Maximum field strength of field i.</td>
<td>Non-negative float (i=1,…,nemax)</td>
<td></td>
</tr>
<tr class="odd">
<td>FIELD_directions</td>
<td>ndir</td>
<td>Number of propagations to perform with different field polarization directions.</td>
<td>Positive integer</td>
<td>Read_theta(i) and Read_phi(i) must be set for i=1,…,ndir. The total number of propagations will be nemax×ndir.</td>
</tr>
<tr class="even">
<td>FIELD_directions</td>
<td>read_theta(i)</td>
<td>Theta coordinate for field polarization direction i</td>
<td>Float from 0–180 (i=1,…,ndir)</td>
<td>Requires dirform=’polar’</td>
</tr>
<tr class="odd">
<td>FIELD_directions</td>
<td>read_phi(i)</td>
<td>Phi coordinate for field polarization direction i</td>
<td>Float from 0–360 (i=1,…,ndir)</td>
<td>Requires dirform=’polar’</td>
</tr>
<tr class="even">
<td>FIELD_directions</td>
<td>read_x(i)</td>
<td>x coordinate for field polarization direction i</td>
<td>Float</td>
<td>Requires dirform=’cart’</td>
</tr>
<tr class="odd">
<td>FIELD_directions</td>
<td>read_y(i)</td>
<td>y coordinate for field polarization direction i</td>
<td>Float</td>
<td>Requires dirform=’cart’</td>
</tr>
<tr class="even">
<td>FIELD_directions</td>
<td>read_z(i)</td>
<td>z coordinate for field polarization direction i</td>
<td>Float</td>
<td>Requires dirform=’cart’</td>
</tr>
<tr class="odd">
<td>SYSTEM_units</td>
<td>dt_units</td>
<td>Units for the ‘dt’ parameter.</td>
<td>‘au’, ‘as’, ‘fs’, ‘ps’</td>
<td>‘au’ atomic units \ ‘as’ attosec \ ‘fs’ femtosec \ ‘ps’ picosec</td>
</tr>
<tr class="even">
<td>SYSTEM_units</td>
<td>eigmax_units</td>
<td>Units for the ‘eigmax’ parameter.</td>
<td>‘au’, ‘eV’</td>
<td>‘au’ atomic units \ ‘eV’ electron volts</td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>dt</td>
<td>Size of the propagation timestep</td>
<td>Positive float</td>
<td></td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>eigmax</td>
<td>Ignore states with energies above eigmax+ionization</td>
<td>Positive float</td>
<td></td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>ionization</td>
<td>estimated ionization energy</td>
<td>Float</td>
<td>if ionization = -1.0, use -HOMO energy</td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>jobtype</td>
<td>Selects the type of propagation to be performed</td>
<td>‘cis’, ‘cisip’, ‘soc’, ‘socip’</td>
<td>‘cis’: TD-CIS \ ‘cisip’: TD-CISDIP \ ‘soc’: TD-CIS with spin-orbit coupling \ ‘socip’: TD-CISDIP with spin-orbit coupling</td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>nstep</td>
<td>Number of timesteps to be propagated</td>
<td>Positive integer</td>
<td></td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>outstep</td>
<td>Output data will only be written on step numbers divisible by outstep</td>
<td>Positive integer</td>
<td>If nstep=50 and outstep=10, data will be written on steps 10, 20, 30, 40, and 50.</td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>nactive</td>
<td>Number of active occupied orbitals (lower orbitals are ignored)</td>
<td>positive integer</td>
<td>nactive = 0 to use all occupied orbitals</td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>nvirtual</td>
<td>Number of active virtual orbitals (higher orbitals are ignored)</td>
<td>positive integer</td>
<td>nvirtual = 0 to use all virtual orbitals</td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>socfac</td>
<td>Spin-orbit coupling scale factor</td>
<td>float</td>
<td>socfac = 1.0 for full SOC \ socfac = 0.0 to turn off SOC</td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>socfacZ</td>
<td>Scales z-component of spin-orbit coupling</td>
<td>N/A</td>
<td></td>
</tr>
<tr class="odd">
<td>SYSTEM</td>
<td>IP_alpha_beta</td>
<td>Which electrons to ionize</td>
<td>.true. or .false.</td>
<td>.true. ionize both alpha and beta \ .false. ionize only beta electrons</td>
</tr>
<tr class="even">
<td>SYSTEM</td>
<td>QeigenDC</td>
<td>use divide-and-conquer eigenvalue solver</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="odd">
<td>InOutFILES</td>
<td>Qread_tdcidata</td>
<td>Type of Gaussian data file to read</td>
<td>.true. or .false.</td>
<td>For legacy TDCI.dat files, use .true. \ For new MatrixElements.faf files, use .false.</td>
</tr>
<tr class="even">
<td>InOutFILES</td>
<td>tdcidatfile</td>
<td>Specifies the filename of the Gaussian data file</td>
<td>String</td>
<td></td>
</tr>
<tr class="odd">
<td>InOutFILES</td>
<td>outputfile</td>
<td>Name of output log file</td>
<td>String</td>
<td></td>
</tr>
<tr class="even">
<td>InOutFILES</td>
<td>restartbinfile</td>
<td>Name of the restart file</td>
<td>String</td>
<td>Setting parameter ‘restart’ to .true. will read from this file. \ Setting parameter ‘Qsave’ to .true. will write this file.</td>
</tr>
<tr class="odd">
<td>InOutFILES</td>
<td>Qwrite_ion_coeff</td>
<td>For sequential double ionization, writes the ion coefficients</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="even">
<td>InOutFILES</td>
<td>Qread_ion_coeff</td>
<td>For sequential double ionization, reads the ion coefficients</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="odd">
<td>InOutFILES</td>
<td>Qmo_dens</td>
<td>Writes the density matrix to MO_density files</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="even">
<td>InOutFILES</td>
<td>verbosity</td>
<td>How chatty should the OUTPUT log file be? Higher=More debug text</td>
<td>Integer</td>
<td>default=2</td>
</tr>
<tr class="odd">
<td>InOutFILES</td>
<td>datfile_enable</td>
<td>Enable writing the plaintext RESULTS, POP, ION dat files.</td>
<td>.true. or .false.</td>
<td>default=.true.</td>
</tr>
<tr class="even">
<td>hdf5</td>
<td>h5inc_enable</td>
<td>Enables h5 file output</td>
<td>.true. or .false.</td>
<td>default=.true.</td>
</tr>
<tr class="odd">
<td>hdf5</td>
<td>h5inc_density</td>
<td>Write density matrix to h5 file at each timestep</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="even">
<td>hdf5</td>
<td>h5inc_psi</td>
<td>Write CI in state basis to h5 file at each timestep</td>
<td>.true. or .false.</td>
<td></td>
</tr>
<tr class="odd">
<td>hdf5</td>
<td>h5inc_psi_det0</td>
<td>Write CI in determinant basis to h5 file at each timestep</td>
<td>.true. or .false.</td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="output-files">Output Files</h2>
<p>This section describes the main output files generated by the TDCI code. Each file (aside from OUTPUT) is named according to the electric-field strength index (X) and direction index (Y), typically appearing in the filename as -eX-dY. For example, the RESULTS file for the progataion with the first specified field and direction will be named RESULTS-e1-d1.dat</p>
<h3 id="output">OUTPUT</h3>
<p>This is the main log file. Similar to a Gaussian log file, it contains both data tables and information about the progress of the calculatiAfter running a TDCI calculation, this should be the first place you check to make sure that it completed successfully. The parameters from ‘input’ and various control parameters are listed at the beginning of the file. OUTPUT also lists the molecule specification, dipole moment, number of basis functions, occupied orbitals, vitrual orbitals and states, specifications for the laser pulse, propagation time step, simulation time, etc. For ‘cis’ or ‘soc’ calculations, energies and coefficients of the ionized state and the neutral state are listed next (for ‘ip’ or ‘socip’, the energies and coefficients are for the doubly ionized state and the singly ionized state). The coefficients of the CI determinants are C(i,a) for ‘cis’ and ‘soc’, and C(i) and C(i,j,a) for ‘ip’ and ‘socip’. The indices are negative for alpha electrons, positive for beta electrons. Abs(i) runs from 1 to the number of occupied orbitals; abs(a) runs from 1 to the number of virtual orbitals. A summary is printed for each individual propagation and includes the maximum field strength and direction, the final norm and dipole moment of the wavefunction. In a good ionization simulation, you should aim for a final norm between 0.3 and 0.6 in the direction with most ionization. You may have to play around with the Emax to achieve this.</p>
<h3 id="results-ex-dy.dat">RESULTS-eX-dY.dat</h3>
<p>The RESULTS file tabulates values of interest at each timestep (the first 3 lines describe the contents). It provides time, field, total norm, ionization rate, and dipole moment components (<code>mu_x,y,z</code>). The MO99 number describes the total number of orbitals needed to contribute 99% of the total ionization rate. NO99 is the same for a user-supplied set of orbitals. The rate data must be divided by norm2 to get the rate for a normalized wavefunction.</p>
<h3 id="pop-ex-dy.dat">POP-eX-dY.dat</h3>
<p>Tracks the populations of occupied orbitals at each timestep (the first 2 lines describe the contents). Also contains the ionization rate for the normalized wavefunction partitioned into contributions from each occupied or virtual orbital, as described in equation <a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a>.</p>
<h3 id="ion-ex-dy.dat">ION-eX-dY.dat</h3>
<p>Tracks the populations of orbitals and states in the ionized system (the first line describe the contents).</p>
<h3 id="data.h5">data.h5</h3>
<p>Requires <code>h5inc_enable = .true.</code> Contains hdf5 group “metadata” for system specific data (dipole moments, CMO coefficients, determinant indexing arrays). Each propagation direction has a group “direction_i” where i is the direction index, which contains subgroups “field_j” where j is the field strength index. Each propagation subgroup “direction_i/field_j/” contains the same data contained in the RESULTS, POP, and ION files. Based on input flags, the MO density matrix and CI vector (in determinant or state basis) at each timestep may be written.</p>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p><a href="https://doi.org/10.1063/1.4874156">Krause, P.; Sonk, J. A.; Schlegel, H. B., Strong field ionization rates simulated with time-dependent configuration interaction and an absorbing potential. J. Chem. Phys. 2014, 140, 174113. 10.1063/1.4874156</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/369.pdf">Mirror</a>)<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p><a href="https://doi.org/10.1021/acs.jpclett.5b00929">Krause, P.; Schlegel, H. B., Angle-dependent ionization of small molecules by time-dependent configuration interaction and an absorbing potential. J. Phys. Chem. Lett. 2015, 6, 2140-2146. 10.1021/acs.jpclett.5b00929</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/381.pdf">Mirror</a>)<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p><a href="https://doi.org/10.1063/5.0034807">Lee, M. K.; Hoerner, P.; Li, W.; Schlegel, H. B., Effect of spin-orbit coupling on strong field ionization simulated with time-dependent configuration interaction. J. Chem. Phys. 2020, 153. 10.1063/5.0034807</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/435.pdf">Mirror</a>)<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p><a href="https://doi.org/10.1021/acs.jpca.7b06108">Hoerner, P.; Schlegel, H. B., Angular dependence of strong field ionization of CH3X (X = F, Cl, Br, or I) using time-dependent configuration interaction with an absorbing potential. J. Phys. Chem. A 2017, 121, 5940-5946. 10.1021/acs.jpca.7b06108</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/409.pdf">Mirror</a>)<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p><a href="https://doi.org/10.1021/acs.jpcc.8b00619">Hoerner, P.; Schlegel, H. B., Angular dependence of strong field ionization of haloacetylenes, HCCX (X = F, Cl, Br, I) using time-dependent configuration interaction with an absorbing potential. J. Phys. Chem. C 2018, 122, 13751–13757. 10.1021/acs.jpcc.8b00619</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/416.pdf">Mirror</a>)<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p><a href="https://doi.org/10.1063/1.5133659">Lee, M. K.; Li, W.; Schlegel, H. B., Angular dependence of strong field sequential double ionization for neon and acetylene simulated with time-dependent configuration interaction using CIS and CISD-IP. J. Chem. Phys. 2020, 152, 064106. 10.1063/1.5133659</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/429.pdf">Mirror</a>)<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7" role="doc-endnote"><p><a href="https://doi.org/10.3389/fchem.2022.866137">Schlegel, H. B.; Hoerner, P.; Li, W., Ionization of HCCI neutral and cations by strong laser fields simulated with time dependent configuration interaction. Front. Chem. 2022, 10, 866137. 10.3389/fchem.2022.866137</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/445.pdf">Mirror</a>)<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8" role="doc-endnote"><p><a href="https://doi.org/10.1063/1.5133659">Lee, M. K.; Li, W.; Schlegel, H. B., Angular dependence of strong field sequential double ionization for neon and acetylene simulated with time-dependent configuration interaction using CIS and CISD-IP. J. Chem. Phys. 2020, 152, 064106. 10.1063/1.5133659</a> – (<a href="https://schlegelgroup.wayne.edu/Pub_folder/429.pdf">Mirror</a>)<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
</body>
</html>
